doc/guide/go_quickstart.hpp

```diff
@@ -0,0 +1,241 @@
+/**
+ * @file go_quickstart.hpp
+ * @author Yashwant Singh Parihar
+
+@page go_quickstart mlpack in Go quickstart guide
+
+@section go_quickstart_intro Introduction
+
+This page describes how you can quickly get started using mlpack from Go and
+gives a few examples of usage, and pointers to deeper documentation.
+
+This quickstart guide is also available for
+@ref cli_quickstart "the command-line", @ref python_quickstart "Python"
+and @ref julia_quickstart "Julia".
+
+@section go_quickstart_install Installing mlpack
+
+Installing the mlpack bindings for Go is somewhat ticker; you can download
+
+@code{.sh}
+go get -d mlpack.org/v1/mlpack
+make install
+@endcode
+
+Building the Go bindings from scratch is a little more in-depth, though.  For
+information on that, follow the instructions on the @ref build page, and be sure
+to specify @c -DBUILD_GO_BINDINGS=ON to CMake;
+
+@section go_quickstart_example Simple mlpack quickstart example
+
+As a really simple example of how to use mlpack from Go, let's do some
+simple classification on a subset of the standard machine learning @c covertype
+dataset.  We'll first split the dataset into a training set and a testing set,
+then we'll train an mlpack random forest on the training data, and finally we'll
+print the accuracy of the random forest on the test dataset.
+
+You can copy-paste this code directly into main.go to run it.
+@code{.go}
+package main
+
+import (
+  "github.com/mlpack.org/v1/mlpack"
+  "fmt"
+  "os"
+)
+func main() {
+
+  // Download dataset.
+  mlpack.DownloadFile("https://www.mlpack.org/datasets/covertype-small.data.csv.gz",
+                      "data.csv.gz")
+  mlpack.DownloadFile("https://www.mlpack.org/datasets/covertype-small.labels.csv.gz",
+                      "labels.csv.gz")
```

Comment:
Really cool that you were able to put this support together so quickly!  Thank you. :+1:

Reply:
Thank you :)